src/main/java/baritone/pathing/movement/movements/MovementParkour.java

```diff
@@ -120,57 +122,62 @@ public static void cost(CalculationContext context, int x, int y, int z, EnumFac
+                break;
             }
             if (!MovementHelper.fullyPassable(context, destX, y + 3, destZ)) {
-                return;
+                maxJump = i - 1;
+                break;
             }
         }
-        if (maxJump != 4) {
-            return;
-        }
+        // parkour place starts here
         if (!context.allowParkourPlace) {
             return;
         }
-        // time 2 pop off with that dank skynet parkour place
-        int destX = x + 4 * xDiff;
-        int destZ = z + 4 * zDiff;
-        IBlockState toReplace = context.get(destX, y - 1, destZ);
-        double placeCost = context.costOfPlacingAt(destX, y - 1, destZ, toReplace);
-        if (placeCost >= COST_INF) {
-            return;
-        }
-        if (!MovementHelper.isReplaceable(destX, y - 1, destZ, toReplace, context.bsi)) {
-            return;
-        }
-        if (!checkOvershootSafety(context.bsi, destX + xDiff, y, destZ + zDiff)) {
-            return;
-        }
-        for (int i = 0; i < 5; i++) {
-            int againstX = destX + HORIZONTALS_BUT_ALSO_DOWN_____SO_EVERY_DIRECTION_EXCEPT_UP[i].getXOffset();
-            int againstY = y - 1 + HORIZONTALS_BUT_ALSO_DOWN_____SO_EVERY_DIRECTION_EXCEPT_UP[i].getYOffset();
-            int againstZ = destZ + HORIZONTALS_BUT_ALSO_DOWN_____SO_EVERY_DIRECTION_EXCEPT_UP[i].getZOffset();
-            if (againstX == x + xDiff * 3 && againstZ == z + zDiff * 3) { // we can't turn around that fast
-                continue;
+        // check parkour jumps from largest to smallest
+        for (int i = maxJump; i > 1; i--) {
+            int destX = x + i * xDiff;
+            int destZ = z + i * zDiff;
+            IBlockState toReplace = context.get(destX, y - 1, destZ);
+            double placeCost = context.costOfPlacingAt(destX, y - 1, destZ, toReplace);
+            if (placeCost >= COST_INF) {
+            	continue;
             }
-            if (MovementHelper.canPlaceAgainst(context.bsi, againstX, againstY, againstZ)) {
-                res.x = destX;
-                res.y = y;
-                res.z = destZ;
-                res.cost = costFromJumpDistance(4) + placeCost + context.jumpPenalty;
-                return;
+            if (!MovementHelper.isReplaceable(destX, y - 1, destZ, toReplace, context.bsi)) {
+            	continue;
+            }
+            if (!checkOvershootSafety(context.bsi, destX + xDiff, y, destZ + zDiff)) {
+            	continue;
+            }
+            for (int j = 0; j < 5; j++) {
+                int againstX = destX + HORIZONTALS_BUT_ALSO_DOWN_____SO_EVERY_DIRECTION_EXCEPT_UP[j].getXOffset();
+                int againstY = y - 1 + HORIZONTALS_BUT_ALSO_DOWN_____SO_EVERY_DIRECTION_EXCEPT_UP[j].getYOffset();
+                int againstZ = destZ + HORIZONTALS_BUT_ALSO_DOWN_____SO_EVERY_DIRECTION_EXCEPT_UP[j].getZOffset();
+                if (againstX == x + xDiff * 3 && againstZ == z + zDiff * 3) { // we can't turn around that fast
```

Comment:
I believe this `3` might need to be changed. The purpose of this `if` is to prevent Baritone from attempting to spin around in midair and place the parkour-place block behind it, which flags anticheat. (or, at least, it flagged NCP in late 2018 when I tested this).